frontend/src/Layout/PrivateRoute/PrivateRoute.jsx

```
@@ -0,0 +1,25 @@
+import { useContext } from "react";
+import { Navigate, useLocation } from "react-router-dom";
+import PropTypes from "prop-types";
+import { AuthContext } from "../AuthProvider/AuthProvider";
+
+const PrivateRoute = ({ children }) => {
+    const { user, loading } = useContext(AuthContext);
+    const location = useLocation();
+
+    if (loading) {
+        return <span className="loading loading-spinner text-info flex justify-center items-center"></span>;
```

Biome (1.9.4)

[error] JSX elements without children should be marked as self-closing. In JSX, it is valid for any element to be self-closing.

Unsafe fix: Use a SelfClosingElement instead


(lint/style/useSelfClosingElements)